### ISSUE TITLE
How to install dependencies from npm?

### ISSUE THREAD

**Issue by tomwayson** [tildeio/glimmer#447]
I need to figure out how to load libraries from npm in a glimmer app.

For example, in my glimmer app, after running `yarn add esri-loader` and then using an `import` statement in my component, I get the following  error when trying to run `ember s`.

```shell
The Broccoli Plugin: [RollupWithDependencies] failed with:
Error: Could not resolve 'esri-loader'
```

For details, see: https://github.com/tomwayson/esri-glimmer-example/issues/1

**Comment by tomwayson** [comment]
I guess another way to ask this question would be, how would I add https://momentjs.com/ to a glimmer app?

**Comment by wycats** [comment]
This is very close to "just work"ing. I'm nominating this as an extremely high priority thing to get over the finish line.

**Comment by wycats** [comment]
(the fact that it doesn't "just work" is kind of an accident, 😱 )

**Comment by tomwayson** [comment]
I'll try to cook up and share a moment.js example, at least the way I'd expect that to work, just in case my own library (esri-loader) is not published properly (which is entirely possible).

**Comment by tomwayson** [comment]
hm... this does seem to be "just working" w/ moment. I'm going to try a few other libs to see if I can figure out where the problem I'm having is.

**Comment by tomwayson** [comment]
I should say, it "just worked" for moment _after_ I set `"allowSyntheticDefaultImports": true` in my glimmer app's tsconfig.json as per https://momentjs.com/docs/#/use-it/typescript/

**Comment by tomwayson** [comment]
I've created a repo to demonstrate how I'd expect things to work.

So far I just added moment, and while it's true that it did work, there were some oddities. In addition to setting `"allowSyntheticDefaultImports": true` as I said above. I did see build errors until I changed my import syntax. See: https://github.com/tomwayson/glimmer-npm-deps-example/blob/09f281b9169a2654318fbf4d99f085c52cb0dcbd/src/ui/components/moment-example/component.ts#L2-L14

I'm now working on adding an unbuilt library like date-fns to see if there are any issues there.

**Comment by tomwayson** [comment]
Yeah, date-fns didn't "just work". See https://github.com/tomwayson/glimmer-npm-deps-example/pull/1

So generally speaking, I'm not finding that you can just `yarn add` a module and then `import` and use it w/in your code.

**Comment by tomwayson** [comment]
Update: 

I was able to resolve the issue w/ my own library by having "main" point to a UMD build and "module" point to an es5/esm as suggested [here](http://2ality.com/2017/04/setting-up-multi-platform-packages.html#module-native-features-esm). Here's the PR: https://github.com/tomwayson/esri-loader/pull/17

One thing to note is that I still see `Cannot find module 'esri-loader'` in the terminal output when I run `npm start`, but the library is clearly loaded b/c the app runs fine: https://github.com/tomwayson/esri-glimmer-example/pull/2

**Comment by knownasilya** [comment]
@wycats has the accident been resolved, or is that tracked in another issue?

**Comment by wycats** [comment]
@knownasilya I'd think this would be tracked in https://github.com/glimmerjs/glimmer.js. The VM itself is agnostic to these kinds of concerns.
